scala-fx/src/main/scala/fx/requires.scala

```diff
@@ -0,0 +1,33 @@
+package fx
+
+import scala.compiletime.error
+import scala.quoted.*
+
+/**
+ * Super-simple compile time requirements.
+ *
+ * If the predicate fails, the compile time error message will print as an error.
+ *
+ * ### Usage
+ * {{{
+ * import fx.requires
+ * opaque type MyInt = Int
+ * object MyInt:
+ *   def apply(x: Int): MyInt =
+ *     require(x < 5, "MyInt must be less than 5")
+ *     5
+ * MyInt(3) // compiles
+ * MyInt(7) // fails compilation with "MyInt must be less than 5"
+ * }}}
+ */
+inline def requires[A](assertion: Boolean, errorMessage: String, value: A): A = {
```

Comment:
nice utility!

Reply:
Yeah, I want a staged version of this so that we can use it both at compile time and runtime.